test/integration/hooks/hooks.test.js

```diff
@@ -310,96 +311,48 @@ describe(Support.getTestDialectTeaser('Hooks'), () => {
         const afterHook = sinon.spy();
-        this.sequelize.beforeBulkSync(() => {
+        const tmpSequelize = createSequelizeInstance();
+
+        tmpSequelize.beforeBulkSync(() => {
           beforeHook();
           throw new Error('Whoops!');
         });
-        this.sequelize.afterBulkSync(afterHook);
 
-        await expect(this.sequelize.sync()).to.be.rejected;
+        tmpSequelize.afterBulkSync(afterHook);
+
+        await expect(tmpSequelize.sync()).to.be.rejectedWith('Whoops!');
         expect(beforeHook).to.have.been.calledOnce;
         expect(afterHook).not.to.have.been.called;
       });
 
-      it('should return an error from after', async function () {
+      it('should return an error from after', async () => {
         const beforeHook = sinon.spy();
         const afterHook = sinon.spy();
+        const tmpSequelize = createSequelizeInstance();
 
-        this.sequelize.beforeBulkSync(beforeHook);
-        this.sequelize.afterBulkSync(() => {
+        tmpSequelize.beforeBulkSync(beforeHook);
+        tmpSequelize.afterBulkSync(() => {
           afterHook();
           throw new Error('Whoops!');
         });
 
-        await expect(this.sequelize.sync()).to.be.rejected;
+        await expect(tmpSequelize.sync()).to.be.rejectedWith('Whoops!');
         expect(beforeHook).to.have.been.calledOnce;
         expect(afterHook).to.have.been.calledOnce;
       });
 
       afterEach(function () {
-        this.sequelize.options.hooks = {};
+        this.sequelize.hooks.removeAllListeners();
       });
-
-    });
-  });
-
-  describe('#removal', () => {
-    it('should be able to remove by name', async function () {
-      const sasukeHook = sinon.spy();
-      const narutoHook = sinon.spy();
-
-      this.User.addHook('beforeCreate', 'sasuke', sasukeHook);
-      this.User.addHook('beforeCreate', 'naruto', narutoHook);
-
-      await this.User.create({ username: 'makunouchi' });
-      expect(sasukeHook).to.have.been.calledOnce;
-      expect(narutoHook).to.have.been.calledOnce;
-      this.User.removeHook('beforeCreate', 'sasuke');
-      await this.User.create({ username: 'sendo' });
-      expect(sasukeHook).to.have.been.calledOnce;
-      expect(narutoHook).to.have.been.calledTwice;
-    });
-
-    it('should be able to remove by reference', async function () {
-      const sasukeHook = sinon.spy();
-      const narutoHook = sinon.spy();
-
-      this.User.addHook('beforeCreate', sasukeHook);
-      this.User.addHook('beforeCreate', narutoHook);
-
-      await this.User.create({ username: 'makunouchi' });
-      expect(sasukeHook).to.have.been.calledOnce;
-      expect(narutoHook).to.have.been.calledOnce;
-      this.User.removeHook('beforeCreate', sasukeHook);
-      await this.User.create({ username: 'sendo' });
-      expect(sasukeHook).to.have.been.calledOnce;
-      expect(narutoHook).to.have.been.calledTwice;
-    });
-
-    it('should be able to remove proxies', async function () {
```

Comment:
Is this not possible anymore?

Reply:
It is but the test is redundant because there is no concept of proxies anymore. Too much boilerplate for something that was used once. I replaced it with a regular hook

Reply:
Sounds good!